clerk/handler_test.go

```diff
@@ -0,0 +1,87 @@
+package clerk_test
+
+import (
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/maticnetwork/heimdall/clerk"
+	"github.com/maticnetwork/heimdall/clerk/types"
+
+	"github.com/maticnetwork/heimdall/app"
+	"github.com/maticnetwork/heimdall/helper/mocks"
+	hmTypes "github.com/maticnetwork/heimdall/types"
+	"github.com/maticnetwork/heimdall/types/simulation"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type HandlerTestSuite struct {
+	suite.Suite
+
+	app    *app.HeimdallApp
+	ctx    sdk.Context
+	cliCtx context.CLIContext
+
+	handler        sdk.Handler
+	contractCaller mocks.IContractCaller
+}
+
+func (suite *HandlerTestSuite) SetupTest() {
+	suite.app, suite.ctx = createTestApp(false)
+	suite.contractCaller = mocks.IContractCaller{}
+	suite.handler = clerk.NewHandler(suite.app.ClerkKeeper, &suite.contractCaller)
+}
+
+func TestHandlerTestSuite(t *testing.T) {
+	suite.Run(t, new(HandlerTestSuite))
+}
+
+func (suite *HandlerTestSuite) TestHandleMsgEventRecord() {
```

Comment:
can this be written as a set of cases? it would reduce duplication of code.

Reply:
same as below